test/jdk/java/lang/StringBuilder/StringBuilderRepeat.java

```diff
@@ -0,0 +1,194 @@
+        sb.repeat("\u2462", 5);
+
+        // multi string UTF16
+
+        sb.repeat("-\u2460", 0);
+        sb.repeat("-\u2461", 1);
+        sb.repeat("-\u2462", 5);
+
+        // CharSequence
+        sb.repeat(MYCHARS, 3);
+
+        // null
+        sb.repeat((String)null, 0);
+        sb.repeat((String)null, 1);
+        sb.repeat((String)null, 5);
+        sb.repeat((CharSequence)null, 0);
+        sb.repeat((CharSequence)null, 1);
+        sb.repeat((CharSequence)null, 5);
+
+
+        String expected = "repeat233333233333-2-3-3-3-3-3\u2461\u2462\u2462\u2462\u2462\u2462\u2461\u2462\u2462\u2462\u2462\u2462-\u2461-\u2462-\u2462-\u2462-\u2462-\u2462abcabcabc" +
+                          "nullnullnullnullnullnullnullnullnullnullnullnull";
+        assertEquals(expected, sb.toString());
+
+        // Codepoints
+
+        sb.setLength(0);
+
+        sb.repeat(0, 0);
+        sb.repeat(0, 1);
+        sb.repeat(0, 5);
+        sb.repeat((int)' ', 0);
+        sb.repeat((int)' ', 1);
+        sb.repeat((int)' ', 5);
+        sb.repeat(0x2460, 0);
+        sb.repeat(0x2461, 1);
+        sb.repeat(0x2462, 5);
+        sb.repeat(0x10FFFF, 0);
+        sb.repeat(0x10FFFF, 1);
+        sb.repeat(0x10FFFF, 5);
+
+        expected = "\u0000\u0000\u0000\u0000\u0000\u0000\u0020\u0020\u0020\u0020\u0020\u0020\u2461\u2462\u2462\u2462\u2462\u2462\udbff\udfff\udbff\udfff\udbff\udfff\udbff\udfff\udbff\udfff\udbff\udfff";
+        assertEquals(expected, sb.toString());
+
+    }
+
+    public void exceptions() {
+        StringBuilder sb = new StringBuilder();
+
+        try {
+            sb.repeat(' ', Integer.MAX_VALUE);
+            throw new RuntimeException("No OutOfMemoryError thrown");
+        } catch (OutOfMemoryError | IndexOutOfBoundsException ex) {
+            // Okay
+        }
+
+        try {
+            sb.repeat("    ", Integer.MAX_VALUE);
+            throw new RuntimeException("No OutOfMemoryError thrown");
+        } catch (OutOfMemoryError | IndexOutOfBoundsException ex) {
+            // Okay
+        }
+
+        try {
+            sb.repeat(MYCHARS, Integer.MAX_VALUE);
+            throw new RuntimeException("No OutOfMemoryError thrown");
+        } catch (OutOfMemoryError | IndexOutOfBoundsException ex) {
+            // Okay
+        }
+
+        try {
+            sb.repeat(' ', -1);
+            throw new RuntimeException("No IllegalArgumentException thrown");
+        } catch (IllegalArgumentException ex) {
+            // Okay
+        }
+
+        try {
+            sb.repeat("abc", -1);
+            throw new RuntimeException("No IllegalArgumentException thrown");
+        } catch (IllegalArgumentException ex) {
+            // Okay
+        }
+
+        try {
+            sb.repeat(MYCHARS, -1);
+            throw new RuntimeException("No IllegalArgumentException thrown");
+        } catch (IllegalArgumentException ex) {
+            // Okay
+        }
+
+        try {
+            sb.repeat(0x10FFFF + 1, -1);
```

Comment:
Should these tests that test invalid codepoints use a valid repeat count as to verify the IAE is correctly thrown for the out-of-bound codepoint? Tests where both params are illegal are fine, too, of course.